internal/types/document_validation_test.go

```diff
@@ -46,6 +47,14 @@ func TestDocumentValidateData(t *testing.T) {
 			doc:    must.NotFail(NewDocument("$v", "bar")),
 			reason: errors.New(`invalid key: "$v" (key must not contain $)`),
 		},
+		"Inf+": {
+			doc:    must.NotFail(NewDocument("v", math.Inf(1))),
+			reason: errors.New(`invalid value: +Inf (infinity values are not allowed)`),
```

Comment:
Let's also log the field that gave an invalid value (so, it'll be easier to debug and just more user-friendly in general).

Reply:
Does it look ok?

Reply:
Yeah, that looks good to me.